engine/access/rpc/backend/connection_factory.go

```diff
@@ -61,16 +68,33 @@ func (cf *ConnectionFactoryImpl) createConnection(address string, timeout time.D
 	return conn, nil
 }
 
+func (cf *ConnectionFactoryImpl) retrieveConnection(address string) (*grpc.ClientConn, error) {
```

Comment:
this could accept a `time.Duration` and pass that into `createConnection`, that way we can maintain the existing timeouts